### PR TITLE
Solution for max product

### PR DESCRIPTION
**Pull request by danielzsh**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._
Solution purposely leaves out details to encourage more thinking, although if reader is truly stuck they can consult implementation

ac submission: https://codeforces.com/gym/100886/submission/245660915

#4165 

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: solutions/gold/cf-100886G.mdx [hunk 1]
```diff
@@ -0,0 +1,69 @@
+---
+id: cf-100886G
+source: CF
+title: Maximum Product
+author: Daniel Zhu
+---
+
+<Spoiler title="Hint">
+</Spoiler>
```
Comment (SansPapyrus683) [inline solutions/gold/cf-100886G.mdx:9]: uh

File: solutions/gold/cf-100886G.mdx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+_Note:_ $D$ is the max number of digits (19).
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+#define sz(v) (int)v.size()
+using ll = long long;
```
Comment (SansPapyrus683) [inline solutions/gold/cf-100886G.mdx:32]: would really appreciate if you didn't use macros

File: solutions/gold/cf-100886G.mdx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	return res;
+}
+
+int main() {
+	string l, r;
+	cin >> l >> r;
+
+	// pad beginning of l with zeros
+	// l = 12, r = 132 -> l = 012, r = 132
+	while (l.size() < r.size()) l.insert(l.begin(), '0');
```
Comment (SansPapyrus683) [inline solutions/gold/cf-100886G.mdx:51]: braces

File: solutions/gold/cf-100886G.mdx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+	// i -> length of LCP (longest common prefix) of str and r
+	bool eq = true;  // whether l[0, i) = r[0, i)
+	for (int i = 0; i <= sz(r); i++) {
+		string cur = r.substr(0, i);
+		eq = eq && l[i] == r[i];
+
+		// in this case, str[i] must equal r[i]
+		// and now the shared prefix has length i + 1
+		// which contradicts our assumption that the LCP has length i
```
Comment (SansPapyrus683) [inline solutions/gold/cf-100886G.mdx:62]:
```cpp
/*
 * yknow, multiline comments
 * are pretty cool
 * in situations like these
 */
```

File: solutions/gold/cf-100886G.mdx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+```cpp
+#include <bits/stdc++.h>
+#define sz(v) (int)v.size()
+using ll = long long;
+
+using namespace std;
+
+// gets product of digits of given string
+// and returns -1 for empty string
```
Comment (SansPapyrus683) [inline solutions/gold/cf-100886G.mdx:37]: when describing functions, i'd like if you used docstrings
```suggestion
/** @return product of the given string's digits (-1 if empty) */
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
W pr

File: solutions/gold/cf-100886G.mdx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using ll = long long;
+using namespace std;
+/** @return product of the given string's digits (-1 if empty) */
+ll prod(string s) {
+	if (!s.length()) return -1;
```
Comment (SansPapyrus683) [inline solutions/gold/cf-100886G.mdx:38]: braces everywehre

**Comment by SansPapyrus683** [comment]
(also did you revert my changes or smth)

**Comment by danielzsh** [comment]
Oh shoot I was just using the usaco.guide github thing and i think it overwrites changes 😭 mb

**Review by SansPapyrus683 — APPROVED** [review]
@ryanchou-dev merge at your recognizance

**Review by ryanchou-dev — APPROVED** [review]
looks good! thanks :)

**Pull request merged**